src/main/java/com/minecrafttas/server/Client.java

```diff
@@ -0,0 +1,278 @@
+	 */
+	public void close() throws IOException {
+		if (this.socket == null || !this.socket.isOpen()) {
+			LOGGER.warn("Tried to close dead socket");
+			return;
+		}
+		
+		this.socket.close();
+	}
+	
+	/**
+	 * Register packet handlers for packets received on the client
+	 */
+	private void registerClientsidePacketHandlers() {
+		// packet 2: tick client
+		this.handlers.put(2, buf -> TickSyncClient.onPacket());
+		
+		// packet 5: change client tickrate
+		this.handlers.put(5, buf -> TASmodClient.tickratechanger.changeClientTickrate(buf.getFloat()));
+	
+		// packet 8: advance tick on clients
+		this.handlers.put(8, buf -> TASmodClient.tickratechanger.advanceClientTick());
+	
+		// packet 9: change tickrate on client
+		this.handlers.put(9, buf -> TASmodClient.tickratechanger.changeClientTickrate(buf.getFloat()));
+	
+		// packet 10: savestate inputs client
+		this.handlers.put(10, buf -> {
+			try {
+				var nameBytes = new byte[buf.getInt()];
+				buf.get(nameBytes);
+				var name = new String(nameBytes);
+				InputSavestatesHandler.savestate(name);
+			} catch (Exception e) {
+				TASmod.LOGGER.error("Exception occured during input savestate: {}", e);
+			}
+		});
+		
+		// packet 11: close GuiSavestateScreen on client
+		this.handlers.put(11, buf -> 
+			Minecraft.getMinecraft().addScheduledTask(() -> {
+				var mc = Minecraft.getMinecraft();
+				if (!(mc.currentScreen instanceof GuiSavestateSavingScreen))
+					mc.displayGuiScreen(new GuiSavestateSavingScreen());
+				else
+					mc.displayGuiScreen(null);
+			})
+		);
+		
+		// packet 12: loadstate inputs client
+		this.handlers.put(12, buf -> {
+			try {
+				var nameBytes = new byte[buf.getInt()];
+				buf.get(nameBytes);
+				var name = new String(nameBytes);
+				InputSavestatesHandler.loadstate(name);
+			} catch (Exception e) {
+				TASmod.LOGGER.error("Exception occured during input loadstate: {}", e);
+			}
+		});
+		
+		// packet 13: unload chunks on client
+		this.handlers.put(13, buf ->
+			Minecraft.getMinecraft().addScheduledTask(() -> 
+				SavestatesChunkControl.unloadAllClientChunks()));
+		
+		// packet 14: request client motion
+		this.handlers.put(14, buf -> {
+			var player = Minecraft.getMinecraft().player;
+			if (player != null) {
+				if (!(Minecraft.getMinecraft().currentScreen instanceof GuiSavestateSavingScreen))
+					Minecraft.getMinecraft().displayGuiScreen(new GuiSavestateSavingScreen());
+				
+				try {
+					// packet 15: send client motion to server
+					TASmodClient.client.write(ByteBuffer.allocate(4 + 8+8+8 + 4+4+4 + 1 + 4).putInt(15)
+						.putDouble(player.motionX).putDouble(player.motionY).putDouble(player.motionZ)
+						.putFloat(player.moveForward).putFloat(player.moveVertical).putFloat(player.moveStrafing)
+						.put((byte) (player.isSprinting() ? 1 : 0))
+						.putFloat(player.jumpMovementFactor)
+					);
+				} catch (Exception e) {
+					TASmod.LOGGER.error("Unable to send packet to server: {}", e);
+				}
+			}
+		});
+	}
```

Comment:
alright so waste two classes which the system was designed to purposefully not have to waste anymore, got it...

yeah sorry I absolutely despise this idea. I designed the system to use anonymous classes/lambdas and you're going to use my really smart implementation but make it super hard to read by adding 2 more classes and defeat it's main purpose too.

I'm generally okay with changing the package "id" to something else but it will still be an integer, period. Feel free to cast an enum around it with a random integer in it's class but DO NOT use strings or anything of that sort.

Reply:
My goal is to group parts of the code, that do shit for the same thing, **together**. My goal is to put everything related to savestates into *one* class, be it events or network handlers. Gets rid of the packet classes, i thought Pancake is happy...

And just so you know it, I am not going for the "least classes possible" way... Just less classes than now... And I think I can still use lambdas in that system to register things if its really small, like /folder, just not for the more complicated stuff that i want to keep in the classes where it belongs to.

I was also thinking about the enum stuff using it as an id, but didn't write it in the comment

Reply:
Ok, i can live with that